src/Analyser/TokenInLayerAnalyser.php

```diff
@@ -23,7 +22,7 @@ class TokenInLayerAnalyser
     private array $tokenTypes;
 
     /**
-     * @param array{types?: array<string>} $config
+     * @param array{types: array<string>} $config
```

Comment:
In theory, we can pass an empty array here, so the original doc block was correct. This just never happens with our current configuration, but might change in the future.

Reply:
In theory, true. Now the expectation is that you cannot do that. It simplifies the code a lot.